src/hal/src/command/mod.rs

```diff
@@ -604,6 +604,46 @@ pub trait CommandBuffer<B: Backend>: fmt::Debug + Any + Send + Sync {
     /// Requests a timestamp to be written.
     unsafe fn write_timestamp(&mut self, stage: pso::PipelineStage, query: query::Query<B>);
 
+    /// TODO docs
+    /// `build_range_infos` must be the same length as `infos` and each element must have a length equal to the parallel `info` entry's `geometries.len()`. there's probably a way to make this safer without mangling the api shape too much...
```

Comment:
These are some spicy requirements, but the API is unsafe, so I guess it could be argued to be reasonable?

Reply:
If they need to be the same length, and we aren't passing the pointers to these arrays directly in any of the backends anyway, why not make this one array? I.e.
```rust
infos: &[(acceleration_structure::AccelerationStructureGeometryDesc<B>, &[acceleration_structure::AccelerationStructureBuildRangeDesc])],
```

Reply:
Also, we should probably make this an iterator, like in other commands (see `wait_events`)

Reply:
Any suggestions on the extra requirement (also for the indirect variant) of:

```rust
for (info, build_range_infos) in infos.iter().zip(build_range_infos.iter()) {
    assert_eq!(info.geometries.len(), build_range_infos.len());
}
```

Reply:
an `assert_eq` sounds ok, for now at least.